core/modules/exact_match.cc

```diff
@@ -241,9 +235,11 @@ void ExactMatch::ProcessBatch(bess::PacketBatch *batch) {
     return pkt->head_data<uint8_t *>() + f.offset;
   };
   table_.MakeKeys(batch, buffer_fn, keys);
-  table_.Find(keys, out_gates, cnt, default_gate);
 
-  RunSplit(out_gates, batch);
+  for (int i = 0; i < batch->cnt(); i++) {
```

Comment:
Unfortunately the compiler doesn't seem to do very well with `batch->cnt()`. Manually doing `int cnt=batch->cnt();` before the loop is much faster, at least on my machine. You may want to check other modules as well.